File: dataset_reduzido/avatar/Python/Code/codeforces_154_A.py
s = input ( )
n = int ( input ( ) )
ans = 0
for i in range ( n ) :
    x = input ( )
    a , b = 0 , 0
    for j in s :
        if j == x [ 0 ] :
            a += 1
        elif j == x [ 1 ] :
            b += 1
        else :
            ans = ans + min ( a , b )
            a , b = 0 , 0
    ans = ans + min ( a , b )
print ( ans )

File: output/LLaMa/avatar/Python/Java/codeforces_154_A.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        int n = Integer.parseInt(scanner.nextLine());
        int ans = 0;

        for (int i = 0; i < n; i++) {
            String x = scanner.nextLine();
            int a = 0, b = 0;
            for (int j = 0; j < s.length(); j++) {
                char c = s.charAt(j);
                if (c == x.charAt(0)) {
                    a++;
                } else if (c == x.charAt(1)) {
                    b++;
                } else {
                    ans += Math.min(a, b);
                    a = 0;
                    b = 0;
                }
            }
            ans += Math.min(a, b);
        }

        System.out.println(ans);
    }
}